packages/sdk/src/utils/EventSubscription.ts

```diff
@@ -0,0 +1,165 @@
+import { DocumentAction } from '../types/ActionTypes';
+import { AnimationPlaybackType, FrameAnimationType } from '../types/AnimationTypes';
+import { BarcodeFrameValidationResult } from '../types/BarcodeTypes';
+import { CharacterStyle } from '../types/CharacterStyleTypes';
+import { DocumentColor } from '../types/ColorStyleTypes';
+import { ConfigType, RuntimeConfigType, SelectedLayoutFrame, Id, AsyncError } from '../types/CommonTypes';
+import { AuthRefreshRequest, AuthCredentials, ConnectorEvent, ConnectorInstance } from '../types/ConnectorTypes';
+import { UndoState } from '../types/DocumentTypes';
+import { DocumentFontFamily } from '../types/FontTypes';
+import { FrameLayoutType, Frame } from '../types/FrameTypes';
+import { LayoutPropertiesType, MeasurementUnit, LayoutListItemType } from '../types/LayoutTypes';
+import { PageSize } from '../types/PageTypes';
+import { ParagraphStyle } from '../types/ParagraphStyleTypes';
+import { CornerRadiusUpdateModel } from '../types/ShapeTypes';
+import { SelectedTextStyle } from '../types/TextStyleTypes';
+import { Variable } from '../types/VariableTypes';
+import { ViewMode } from '../types/ViewModeTypes';
+import { Viewport } from '../types/ViewportTypes';
+import { ToolType } from './Enums';
+
+class EventSubscriptionBase {}
+
+export class EventSubscription<T extends (...args: unknown[]) => unknown> extends EventSubscriptionBase {
+    private callbacks: Map<string, T> = new Map();
+    legacyEventHandler: (() => T | undefined) | undefined;
+
+    constructor(legacyEventHandler?: () => T | undefined) {
+        super();
+        this.legacyEventHandler = legacyEventHandler;
+    }
+
+    // Register a new callback with an optional name
+    registerCallback(callback: T, name?: string): string {
```

Comment:
when do we might need an optional name useCase?  Wouldn't be simpler to subscribe to the SDK defined events only?

Reply:
The name here is not used for the `eventName`, rather is just a key for the callback that was added. The main purpose is to be able to unsubscribe more easily. Imagine only needing to subscribe for a limited amount of time, for example. Maybe updating the parameter name to `key` would make that more clear?